src/garage/np/algos/batch_polopt.py

```diff
@@ -30,7 +31,9 @@ class BatchPolopt(RLAlgorithm):
 
     """
 
-    def __init__(self, policy, baseline, discount, max_path_length, n_samples):
+    def __init__(self, env_spec, policy, baseline, discount, max_path_length,
+                 n_samples):
+        self.env_spec = env_spec
```

Comment:
where is this used?

Reply:
Every algorithms have `env_spec` so I think it should be in the base class?

Reply:
can you explain why you added this? i don't see where it is used.

Reply:
Seems to me every algorithms have env_spec so I think it should be in the base class?

Reply:
oh, are the algorithms actually using `env_spec`? if so, then yes add it to the base class. if it's just there and not used, then remove it. if only one or two algorithms need it, probably don't include it.

Reply:
How many algorithms do not use env_spec? How many do? When in doubt don't require it. We can improve other parts of the RLAlgorithm interface later.

Reply:
For current implementation, all `torch` and `np` algorithms require it. But none seems to actually use it.

Reply:
TF algorithms use it

Reply:
i don't think you should make it part of the base class in this PR, since it's only needed for TF

Reply:
ok I'll roll it back